recipes/android-ndk/all/conanfile.py

```diff
@@ -0,0 +1,207 @@
+                filename = os.path.join(root, filename)
+                with open(filename, 'rb') as f:
+                    sig = f.read(4)
+                    if type(sig) is str:
+                        sig = [ord(s) for s in sig]
+                    else:
+                        sig = [s for s in sig]
+                    if len(sig) > 2 and sig[0] == 0x23 and sig[1] == 0x21:
+                        self.output.info('chmod on script file: "%s"' % filename)
+                        self._chmod_plus_x(filename)
+                    elif sig == [0x7F, 0x45, 0x4C, 0x46]:
+                        self.output.info('chmod on ELF file: "%s"' % filename)
+                        self._chmod_plus_x(filename)
+                    elif sig == [0xCA, 0xFE, 0xBA, 0xBE] or \
+                         sig == [0xBE, 0xBA, 0xFE, 0xCA] or \
+                         sig == [0xFE, 0xED, 0xFA, 0xCF] or \
+                         sig == [0xCF, 0xFA, 0xED, 0xFE] or \
+                         sig == [0xFE, 0xEF, 0xFA, 0xCE] or \
+                         sig == [0xCE, 0xFA, 0xED, 0xFE]:
+                        self.output.info('chmod on Mach-O file: "%s"' % filename)
+                        self._chmod_plus_x(filename)
+
+    def package(self):
+        self.copy(pattern="*", dst=".", src=self._source_subfolder, keep_path=True, symlinks=True)
+        self.copy(pattern="*", dst="bin", src=self._source_subfolder, keep_path=True, symlinks=True)
+        self.copy(pattern="*NOTICE", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="*NOTICE.toolchain", dst="licenses", src=self._source_subfolder)
+        self.copy("cmake-wrapper.cmd")
+        self.copy("cmake-wrapper")
+        self._fix_permissions()
+
+    @property
+    def _host(self):
+        return self._platform + "-x86_64"
+
+    @property
+    def _ndk_root(self):
+        return os.path.join(self.package_folder, "toolchains", "llvm", "prebuilt", self._host)
+
+    def _tool_name(self, tool):
+        if 'clang' in tool:
+            suffix = '.cmd' if self.settings_build.os == 'Windows' else ''
+            return '%s%s-%s%s' % (self._clang_triplet, self.settings_target.os.api_level, tool, suffix)
+        else:
+            suffix = '.exe' if self.settings_build.os == 'Windows' else ''
+            return '%s-%s%s' % (self._llvm_triplet, tool, suffix)
+
+    def _define_tool_var(self, name, value):
+        ndk_bin = os.path.join(self._ndk_root, 'bin')
+        path = os.path.join(ndk_bin, self._tool_name(value))
+        self.output.info('Creating %s environment variable: %s' % (name, path))
+        return path
+
+    # def package_id(self):
+    #     self.info.include_build_settings()
+    #     del self.info.settings.arch
+    #     del self.info.settings.os.api_level
+
+    @staticmethod
+    def _chmod_plus_x(filename):
+        if os.name == 'posix':
+            os.chmod(filename, os.stat(filename).st_mode | 0o111)
+
+    def package_info(self):
```

Comment:
Shouldn't this package have a `toolchain` method?
See https://docs.conan.io/en/latest/creating_packages/toolchains/cmake.html

Reply:
needs further investigation since I never used that methode, but a good point! 

also, there is actually a toolchain file included in, <NDK>/build/cmake/android.toolchain.cmake
need to check if this could simplify the cmake  wrapper

Reply:
I'm also interested in knowing if this package will be compatible with the new `toolchain` features.
For instance here https://docs.conan.io/en/latest/integrations/cross_platform/android.html#use-built-in-conan-toolchain they use a profile:host with `android_ndk_installer/r20@bincrafters/stable` and conan_toolchain.cmake gets filled with all the information.

Is the current package a drop-in replacement for the bincrafters recipe?

Reply:
@Hopobcn , this NDK package makes much more, it allows also building packages that do not use CMake, like openssl or curl

Why creating an own toolchain if you can get one from the NDK, that is for sure the most complete one? 
If you create your toolchain file, it should just include that one from the NDK anyway

It is not a drop in replacement, because it works with the (not so) new 2 profile approach, the bincrafter one used the one profile way with os_buid and os_arch
The future is the 2 profile way.